crates/iota-light-client/src/checkpoint.rs

```diff
@@ -0,0 +1,492 @@
+        info!("Synced epoch: 0, checkpoint: {last_seq}",);
+        last_seq
+    };
+
+    let object_store = IotaObjectStore::new(config)?;
+    let last_sum = object_store.get_checkpoint_summary(last_seq).await?;
+    let client = IotaClientBuilder::default()
+        .build(config.rpc_url.as_str())
+        .await?;
+    let latest_seq = client
+        .read_api()
+        .get_latest_checkpoint_sequence_number()
+        .await?;
+    let latest_sum = object_store.get_checkpoint_summary(latest_seq).await?;
+    // Sequentially record all the missing end of epoch checkpoints numbers
+    for target_epoch in (last_sum.epoch() + 1)..latest_sum.epoch() {
+        let target_seq = query_last_checkpoint_of_epoch(config, target_epoch).await?;
+        checkpoints_list.checkpoints.push(target_seq);
+        info!("Synced epoch: {target_epoch}, checkpoint: {target_seq}");
+    }
+
+    Ok(checkpoints_list)
+}
+
+pub async fn sync_and_check_checkpoints(config: &Config) -> anyhow::Result<()> {
+    let checkpoints_list = sync_checkpoint_list_to_latest(config)
+        .await
+        .context("Failed to sync checkpoint list")?;
+
+    // Create a list of summaries that need to be downloaded
+    let mut missing = Vec::new();
+    for seq in checkpoints_list.checkpoints.iter().copied() {
+        if !config.checkpoint_summary_file_path(seq, None).exists() {
+            missing.push(seq);
+        }
+    }
+
+    // Load the genesis committee
+    let genesis_committee = Genesis::load(config.genesis_blob_file_path())?
+        .committee()
+        .context("Failed to load genesis file")?;
+
+    if let Some(_checkpoint_store_url) = &config.object_store_url {
+        // Download summaries from checkpoint object store
+        // TODO implement it, blocked by https://github.com/iotaledger/iota/issues/4908
+        warn!("Downloading from a checkpoint store is not supported yet.");
+    }
+
+    if let Some(archive_store_config) = &config.archive_store_config {
+        info!("Downloading missing checkpoints from archive store.");
+
+        // Download summaries from archive store
+        let archive_reader_config = ArchiveReaderConfig {
+            remote_store_config: archive_store_config.clone(),
+            download_concurrency: NonZeroUsize::new(5).unwrap(),
+            use_for_pruning_watermark: false,
+        };
+
+        let metrics = ArchiveReaderMetrics::new(&Registry::default());
+        let archive_reader = ArchiveReader::new(archive_reader_config, &metrics)?;
+        archive_reader.sync_manifest_once().await?;
+        archive_reader
+            .download_summaries_for_list_no_verify(missing, &config.checkpoints_dir)
+            .await?;
+    } else {
+        info!("Downloading missing checkpoints from full node.");
+
+        // Download summaries from the full node
+        let client = iota_rest_api::Client::new(&config.rpc_url);
+
+        // We only need the first 2 end-of-epoch checkpoints for the tests
```

Comment:
What's this comment here about? Don't see anything related to tests and `2` here